desc/objectives/_geometry.py

```diff
@@ -766,12 +830,55 @@ def compute(self, equil_params, surface_params=None, constants=None):
                 profiles={},
             )["x"]
             surface_coords = rpz2xyz(surface_coords)
-        d = safenorm(plasma_coords[:, None, :] - surface_coords[None, :, :], axis=-1)
+
+        diff_vec = plasma_coords[:, None, :] - surface_coords[None, :, :]
+        d = safenorm(diff_vec, axis=-1)
+
+        point_signs = jnp.ones(surface_coords.shape[0])
+        if self._use_signed_distance:
+            surface_coords_rpz = xyz2rpz(surface_coords)
+
+            plasma_coords_rpz = plasma_coords_rpz.reshape(
+                constants["equil_transforms"]["grid"].num_zeta,
+                constants["equil_transforms"]["grid"].num_theta,
+                3,
+            )
+            surface_coords_rpz = surface_coords_rpz.reshape(
+                constants["surface_transforms"]["grid"].num_zeta,
+                constants["surface_transforms"]["grid"].num_theta,
+                3,
+            )
+
+            # loop over zeta planes
+            def fun(plasma_pts_at_zeta_plane, surface_pts_at_zeta_plane):
+                plasma_pts_at_zeta_plane = jnp.vstack(
+                    (plasma_pts_at_zeta_plane, plasma_pts_at_zeta_plane[0, :])
+                )
+
+                surface_pts_at_zeta_plane
```

Comment:
Is there a missing comma here or is surface_pts_at_zeta_planept_sign a single variable?
The formatting looks weird.

Reply:
ah good catch idt this line is doing anything rn